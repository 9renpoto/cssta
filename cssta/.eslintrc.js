module.exports = {
    "extends": "airbnb-base",
    "installedESLint": true,
    "plugins": [
        "react",
        "jsx-a11y",
        "import"
    ],
    "rules": {
        "react/jsx-filename-extension": [0],
        "comma-dangle": ["error", {
            arrays: "always-multiline",
            objects: "always-multiline",
            imports: "always-multiline",
            exports: "always-multiline",
            functions: "never",
        }],
    }
};
