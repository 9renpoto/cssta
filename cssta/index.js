/* global document */
const postcss = require('postcss');
const { transformClassNames } = require('postcss-transform-classes');
const { transformAnimationNames } = require('postcss-transform-animations');

let devId = 0;
const getDevId = () => {
  devId += 1;
  return `css-${devId}`;
};

const createCssElement = (cssText) => {
  const styleElement = document.createElement('style');
  const styleBody = document.createTextNode(cssText);
  styleElement.appendChild(styleBody);
  document.getElementsByTagName('head')[0].appendChild(styleElement);
};

const transform = (css, {
  transformClassName,
  transformAnimationName,
}) => {
  const classNameMap = {};
  const animationNameMap = {};

  const root = postcss.parse(css);

  transformClassNames({
    transform: (value) => {
      if (value in classNameMap) return classNameMap[value];

      const transformValue = transformClassName(value);
      classNameMap[value] = transformValue;
      return transformValue;
    },
  }, root);

  transformAnimationNames({
    transform: (value) => {
      if (value in animationNameMap) return animationNameMap[value];

      const transformValue = transformAnimationName(value);
      animationNameMap[value] = transformValue;
      return transformValue;
    },
  }, root);

  const output = root.toString();

  return { output, classNameMap, animationNameMap };
};

module.exports = (css, prefix = getDevId()) => {
  const { output, classNameMap } = transform(css, {
    transformClassName: className => `${prefix}__${className}`,
    transformAnimationName: animationName => `${prefix}__${animationName}`,
  });
  createCssElement(output);
  return classNameMap;
};
module.exports.transform = transform;
