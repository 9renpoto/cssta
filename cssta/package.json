{
  "name": "cssta",
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "postcss": "^5.2.0",
    "postcss-selector-parser": "^2.2.1"
  },
  "devDependencies": {
    "eslint": "^3.5.0",
    "eslint-config-airbnb": "^11.1.0",
    "eslint-plugin-import": "^1.15.0",
    "eslint-plugin-jsx-a11y": "^2.2.2",
    "eslint-plugin-react": "^6.3.0"
  }
}
