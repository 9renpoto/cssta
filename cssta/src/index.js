/* global document */
const extractRules = require('./extractRules');
const createComponent = require('./createComponent');

let devId = 0;
const getDevId = name => () => {
  devId += 1;
  return `${name}-${devId}`;
};

const createCssElement = (cssText) => {
  const styleElement = document.createElement('style');
  const styleBody = document.createTextNode(cssText);
  styleElement.appendChild(styleBody);
  document.getElementsByTagName('head')[0].appendChild(styleElement);
};

const opts = {
  generateClassName: getDevId('rule'),
  generateAnimationName: getDevId('animation'),
};

const style = tagName => (cssText, ...otherAttributes) => {
  if (otherAttributes.length) {
    throw new Error('You cannot use string interpolation with cssta');
  }

  const { css, baseClassName, classNameMap } = extractRules(cssText, opts);

  createCssElement(css);

  return createComponent(tagName, baseClassName, classNameMap);
};

`a abbr address area article aside audio b base bdi bdo big blockquote body br button canvas
caption cite code col colgroup data datalist dd del details dfn dialog div dl dt em embed fieldset
figcaption figure footer form h1 h2 h3 h4 h5 h6 head header hgroup hr html i iframe img input ins
kbd keygen label legend li link main map mark menu menuitem meta meter nav noscript object ol
optgroup option output p param picture pre progress q rp rt ruby s samp script section select small
source span strong style sub summary sup table tbody td textarea tfoot th thead time title tr track
u ul var video wbr circle clipPath defs ellipse g image line linearGradient mask path pattern
polygon polyline radialGradient rect stop svg text tspan`.split(/\s+/m).forEach((tagName) => {
  style[tagName] = style(tagName);
});

module.exports = style;

console.log(extractRules(`
  :hover {
    color: blue;
  }
`, opts));
