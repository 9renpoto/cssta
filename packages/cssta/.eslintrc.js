module.exports = {
    "extends": "airbnb-base",
    "installedESLint": true,
    "plugins": [
        "import",
        "flowtype"
    ],
    "rules": {
        "react/jsx-filename-extension": [0],
        "flowtype/require-valid-file-annotation": [2, "always"],
        "comma-dangle": ["error", {
            arrays: "always-multiline",
            objects: "always-multiline",
            imports: "always-multiline",
            exports: "always-multiline",
            functions: "never",
        }],
        "spaced-comment": [0],
        "arrow-parens": [0]
    }
};
