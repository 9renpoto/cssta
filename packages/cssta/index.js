// @flow
module.exports = require('./lib/web');
