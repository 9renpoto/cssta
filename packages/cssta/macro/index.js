module.exports = require("./dev.macro");
