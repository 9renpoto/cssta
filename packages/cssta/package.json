{
  "name": "cssta",
  "version": "0.6.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "test:watch": "jest --watch",
    "test:coverage": "jest --coverage",
    "flow": "flow; test $? -eq 0 -o $? -eq 2",
    "lint": "eslint ./src/**/*.js",
    "clean": "npm run clean:lib; npm run clean:vendor",
    "clean:lib": "rm -rf ./lib || :;",
    "clean:vendor": "rm -rf ./vendor || :;",
    "build": "npm run clean; npm run build:lib; npm run build:vendor",
    "build:lib": "babel ./src --out-dir ./lib --ignore __tests__,__mocks__",
    "build:vendor": "webpack --config ./webpack.config.js",
    "prepublish": "npm run build; cp ../../docs/index.md README.md"
  },
  "keywords": [
    "postcss",
    "modules",
    "css-modules",
    "css",
    "minify",
    "min",
    "class",
    "className",
    "react",
    "css-in-js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/jacobp100/cssta"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "css-color-function": "^1.3.0",
    "css-to-react-native": "^2.0.0",
    "dependency-graph": "^0.5.0",
    "events": "^1.1.1",
    "fbjs": "^0.8.6",
    "postcss": "^5.2.6",
    "postcss-selector-parser": "^2.2.2",
    "postcss-transform-animations": "^1.0.1",
    "prop-types": "^15.5.8"
  },
  "devDependencies": {
    "babel-cli": "^6.18.0",
    "babel-preset-es2015": "^6.18.0",
    "eslint": "^3.11.1",
    "eslint-config-airbnb-base": "^10.0.1",
    "eslint-plugin-flowtype": "^2.30.3",
    "eslint-plugin-import": "^2.2.0",
    "flow-bin": "^0.42.0",
    "jest": "^17.0.3",
    "react": "^15.4.1",
    "react-test-renderer": "^15.4.1",
    "webpack": "^1.14.0"
  },
  "peerDependencies": {
    "react": "^15.0.0"
  },
  "jest": {
    "testEnvironment": "node",
    "collectCoverageFrom": [
      "src/**/*.js",
      "!src/index.js",
      "!src/web/index.js",
      "!src/packages/**/*.js"
    ]
  }
}
