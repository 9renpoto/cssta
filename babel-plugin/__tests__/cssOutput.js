const path = require("path");
const fs = require("fs");
const { transform } = require("babel-core");
const plugin = require("..");

const run = (input, filename, options) => {
  const { code } = transform(input, {
    plugins: [[plugin, { output: filename, preserveCssFile: true }]],
    babelrc: false,
    ...options
  });

  return code;
};

const teardown = filename => {
  const css = fs.readFileSync(filename, "utf-8");
  fs.unlinkSync(filename);
  return css;
};

test("Css output", () => {
  const filename = path.join(__dirname, "output1.css");
  plugin.resetGenerators();
  const jsOutput = run(
    `
      import cssta from 'cssta';

      const Component = cssta.button\`
        color: red;
      \`;
    `,
    filename
  );
  expect(jsOutput).toMatchSnapshot();
  expect(teardown(filename)).toMatchSnapshot();
});

test("Css output with multiple files", () => {
  const cssFilename = path.join(__dirname, "output2.css");
  plugin.resetGenerators();
  const jsOutput1 = run(
    `
      import cssta from 'cssta';

      const Component = cssta.button\`
        color: red;
      \`;
    `,
    cssFilename,
    { filename: "input1.js" }
  );
  const jsOutput2 = run(
    `
      import cssta from 'cssta';

      const Component = cssta.button\`
        color: blue;
      \`;
    `,
    cssFilename,
    { filename: "input2.js" }
  );

  expect(jsOutput1).toMatchSnapshot();
  expect(jsOutput2).toMatchSnapshot();
  expect(teardown(cssFilename)).toMatchSnapshot();
});

test("Css output overwriting files", () => {
  const cssFilename = path.join(__dirname, "output2.css");
  plugin.resetGenerators();
  const jsOutput1 = run(
    `
      import cssta from 'cssta';

      const Component = cssta.button\`
        color: red;
      \`;
    `,
    cssFilename,
    { filename: "input1.js" }
  );
  const jsOutput2 = run(
    `
      import cssta from 'cssta';

      const Component = cssta.button\`
        color: blue;
      \`;
    `,
    cssFilename,
    { filename: "input1.js" }
  );

  expect(jsOutput1).toMatchSnapshot();
  expect(jsOutput2).toMatchSnapshot();
  expect(teardown(cssFilename)).toMatchSnapshot();
});

test("Css output overwriting files with multiple files", () => {
  const cssFilename = path.join(__dirname, "output3.css");
  plugin.resetGenerators();
  const jsOutput1 = run(
    `
      import cssta from 'cssta';
      const Component = cssta.button\`color: red;\`;
    `,
    cssFilename,
    { filename: "input1.js" }
  );
  run(
    `
      import cssta from 'cssta';
      const Component = cssta.button\`color: yellow;\`;
    `,
    cssFilename,
    { filename: "input2.js" }
  );
  const jsOutput3 = run(
    `
      import cssta from 'cssta';
      const Component = cssta.button\`color: green;\`;
    `,
    cssFilename,
    { filename: "input3.js" }
  );
  const jsOutput2 = run(
    `
      import cssta from 'cssta';
      const Component = cssta.button\`color: blue;\`;
    `,
    cssFilename,
    { filename: "input2.js" }
  );

  expect(jsOutput1).toMatchSnapshot();
  expect(jsOutput2).toMatchSnapshot();
  expect(jsOutput3).toMatchSnapshot();
  // Should not move contents of input2
  expect(teardown(cssFilename)).toMatchSnapshot();
});
