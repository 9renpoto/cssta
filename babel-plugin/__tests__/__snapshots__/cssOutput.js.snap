// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Css output 1`] = `
"import _createComponent from 'cssta/lib/web/createComponent';


const Component = _createComponent('button', null, {
  'defaultClassName': 'A',
  'classNameMap': {}
});"
`;

exports[`Css output 2`] = `
"/* File generated by babel-plugin-cssta */

/* Start unknown */
.A {
    color: red
}
/* End unknown */
"
`;

exports[`Css output overwriting files 1`] = `
"import _createComponent from 'cssta/lib/web/createComponent';


const Component = _createComponent('button', null, {
  'defaultClassName': 'A',
  'classNameMap': {}
});"
`;

exports[`Css output overwriting files 2`] = `
"import _createComponent from 'cssta/lib/web/createComponent';


const Component = _createComponent('button', null, {
  'defaultClassName': 'B',
  'classNameMap': {}
});"
`;

exports[`Css output overwriting files 3`] = `
"/* File generated by babel-plugin-cssta */

/* Start input1.js */
.B {
    color: blue
}
/* End input1.js */
"
`;

exports[`Css output overwriting files with multiple files 1`] = `
"import _createComponent from 'cssta/lib/web/createComponent';

const Component = _createComponent('button', null, {
      'defaultClassName': 'A',
      'classNameMap': {}
});"
`;

exports[`Css output overwriting files with multiple files 2`] = `
"import _createComponent from 'cssta/lib/web/createComponent';

const Component = _createComponent('button', null, {
      'defaultClassName': 'D',
      'classNameMap': {}
});"
`;

exports[`Css output overwriting files with multiple files 3`] = `
"import _createComponent from 'cssta/lib/web/createComponent';

const Component = _createComponent('button', null, {
      'defaultClassName': 'C',
      'classNameMap': {}
});"
`;

exports[`Css output overwriting files with multiple files 4`] = `
"/* File generated by babel-plugin-cssta */

/* Start input1.js */
.A {
    color: red
}
/* End input1.js */

/* Start input2.js */
.D {
    color: blue
}
/* End input2.js */

/* Start input3.js */
.C {
    color: green
}
/* End input3.js */
"
`;

exports[`Css output with multiple files 1`] = `
"import _createComponent from 'cssta/lib/web/createComponent';


const Component = _createComponent('button', null, {
  'defaultClassName': 'A',
  'classNameMap': {}
});"
`;

exports[`Css output with multiple files 2`] = `
"import _createComponent from 'cssta/lib/web/createComponent';


const Component = _createComponent('button', null, {
  'defaultClassName': 'B',
  'classNameMap': {}
});"
`;

exports[`Css output with multiple files 3`] = `
"/* File generated by babel-plugin-cssta */

/* Start input1.js */
.A {
    color: red
}
/* End input1.js */

/* Start input2.js */
.B {
    color: blue
}
/* End input2.js */
"
`;
